app/src/main/java/com/asma/tasky/core/presentation/components/Navigation.kt

```diff
@@ -49,13 +56,64 @@ fun Navigation(
         }
 
         composable(Screen.TaskScreen.route) {
-            TaskScreen()
+            // todo check if a task object is passed from Agenda Screen
+            // todo also check if should show in edit mode
+
+            var title = stringResource(R.string.new_task)
+            var description = stringResource(R.string.task_description)
+
+            if (navController.currentBackStackEntry!!.savedStateHandle.contains(Constants.KEY_TITLE)) {
+                title =
+                    navController.currentBackStackEntry!!.savedStateHandle.get<String>(
+                        Constants.KEY_TITLE
+                    ) ?: stringResource(R.string.new_task)
+            }
+
+            if (navController.currentBackStackEntry!!.savedStateHandle.contains(Constants.KEY_DESCRIPTION)) {
+                description =
+                    navController.currentBackStackEntry!!.savedStateHandle.get<String>(
+                        Constants.KEY_DESCRIPTION
+                    ) ?: stringResource(R.string.task_description)
+            }
+
+            // default task if no task object passed
+            val editable = true
+            val task = Task(
+                title = title,
+                description = description,
+                id = "",
+                startDate = System.currentTimeMillis()
+            )
```

Comment:
These default values are all considered UI state and should therefore be in the default state of the ViewModel for that screen